src/components/reportPage/ReportPage.jsx

```diff
@@ -3,10 +3,10 @@ import ProgressBar from "react-bootstrap/ProgressBar";
 import { useLocation } from "react-router";
 
 import "./results-page.css";
-import Repository from "../data/Repository";
+import Repository from "../../data/Repository";
 
 const ReportPage = () => {
-  const { files, email } = useLocation().state;
+  const { files, email } = useLocation().state ?? {};
```

Comment:
what does this do ` ?? {}`

Reply:
`??` is a [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator).
`{}` instantiates an empty object.
Together: if `useLocation().state` is null, then use an empty object to assign `files` and `email` via [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) (files and email will then be null). Without `?? {}` there is an error because you are trying to destructure `null`, which isn't an object